sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -177,28 +177,29 @@ private[hive] object SparkSQLCLIDriver extends Logging {
       sessionState.info = new PrintStream(System.err, true, UTF_8.name())
       sessionState.err = new PrintStream(System.err, true, UTF_8.name())
     } catch {
-      case e: UnsupportedEncodingException => System.exit(3)
+      case _: UnsupportedEncodingException => System.exit(3)
     }
 
-    if (sessionState.database != null) {
+    if (dummySessionState.database != null) {
```

Comment:
why we still use `dummySessionState` here while `sessionState` is already created?

Reply:
because of the HiveConf instance which `sessionState` uses to initialize itself need a full scope of all configurations including spark confs, hadoop confs, hive-site.xml, and command-line properties.  the command-line properties need be gen first.

Reply:
the dummy one has parsed -e -database -f e.t.c. these info don't transfer